t/batches.t

```diff
@@ -201,11 +223,33 @@ subtest 'batch with several domains' => sub {
 
     is( $res, 1, 'correct batch job id returned' );
 
+    # "get_batch_job_result deprecated to be removed by v2025.2
     $res = $rpcapi->get_batch_job_result( { batch_id => 1 } );
 
     is( $res->{nb_running}, @domains, 'correct number of runninng tests' );
     is( $res->{nb_finished}, 0, 'correct number of finished tests' );
-
+    
+    # No lists of test IDs requested
+    $res = $rpcapi->batch_status( { batch_id => 1 } );
+
+    is( $res->{waiting_count}, @domains, 'correct number of runninng tests' );
+    is( $res->{running_count}, 0, 'correct number of finished tests' );
+    is( $res->{finished_count}, 0, 'correct number of finished tests' );
+    is( $res->{waiting_tests}, undef, 'correct undefined list of waiting tests' );
+    is( $res->{running_tests}, undef, 'correct undefined list of running tests' );
+    is( $res->{finished_tests}, undef, 'correct undefined list of finished tests' );
+
+    # List of waiting test IDs requested
+    $res = $rpcapi->batch_status( { batch_id => 1, list_waiting_tests => 1 } );
+
+    is( $res->{waiting_count}, @domains, 'correct number of runninng tests' );
+    is( $res->{running_count}, 0, 'correct number of finished tests' );
+    is( $res->{finished_count}, 0, 'correct number of finished tests' );
+    is( scalar @{ $res->{waiting_tests} }, scalar @domains, 'correct number of waiting tests in list' );
```

Comment:
Maybe use `eq_or_diff` here to check the actual contents instead of just the amount?

Reply:
Updated.